src/de/espend/idea/laravel/controller/namespace/LaravelControllerNamespaceCutter.java

```diff
@@ -0,0 +1,91 @@
+package de.espend.idea.laravel.controller.namespace;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiElement;
+import com.jetbrains.php.PhpIndex;
+import com.jetbrains.php.lang.psi.elements.Field;
+import com.jetbrains.php.lang.psi.elements.PhpClass;
+import de.espend.idea.laravel.LaravelSettings;
+import fr.adrienbrault.idea.symfony2plugin.codeInsight.utils.PhpElementsUtil;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+import java.util.HashSet;
+
+public class LaravelControllerNamespaceCutter implements ControllerNamespaceCutter {
+
+    private String prefix;
+
+    /**
+     * Collection of possible default namespaces. default namespace + default namespace with prefix
+     */
+    private Collection<String> defaultNamespaces = new HashSet<>();
+
+    @Override
+    public void init(@NotNull Project project, @Nullable String prefix) {
```

Comment:
please make this immutable or move to constructor. as we living in a multithread application with multiple projects this will conflict each other